src/go/rpk/pkg/tuners/network/nics.go

```diff
@@ -35,14 +35,20 @@ func getDefaultMode(
 			return "", err
 		}
 
-		// Currently we use only the mq mode because the idea behind sq and sq-split modes is that
-		// a core is *dedicated* to IRQs (i.e., Redpanda core does not run on that core or lcore),
-		// but we don't currently support propagating a cpuset to Redpanda, so it will run on the IRQ
-		// core, causing it to be the primary bottleneck.
-		zap.L().Sugar().Debugf("Using mq mode (hardcoded) for '%s': '%d' cores, '%d' PUs and '%d' rx queues",
-			nic.Name(), numOfCores, numOfPUs, rxQueuesCount)
-
-		return irq.Mq, nil
+		// TODO: check
+		//   - no cpuset specified in config
+		//   - --smp is compatible
+		var mode irq.Mode
+		if numOfPUs >= 16 {
+			mode = irq.Dedicated
+		} else {
+			mode = irq.Mq
+		}
+
+		zap.L().Sugar().Debugf("Using '%s' mode for '%s': '%d' cores, '%d' PUs and '%d' rx queues",
+			mode, nic.Name(), numOfCores, numOfPUs, rxQueuesCount)
+
+		return mode, nil
 	}
 
 	if nic.IsBondIface() {
```

Comment:
How about this case? Too rare to care about?

Reply:
We could at least have a log line the one above in this case so when we look at the logs we don't have to wonder like "oh was this a bond interface (if we don't see the expected line)".